src/redemptions/redemption.service.ts

```diff
@@ -255,6 +279,7 @@ export class RedemptionService {
         id_details: instanceToPlain(data.idDetails),
         failure_message: data.failureMessage,
         public_address: data.publicAddress,
```

Comment:
If add age as param that is `undefined` prisma trys to use `NaN` and fails. Must omit property if not defined

Reply:
This seems odd... maybe a prisma bug?

Reply:
Yea I am guessing thats the case, I saw some open tickets in prisma relative to this, just wrote it this way as workaround.
The exact error was as follows (when age was undefined)

```
→ 275   return client.redemption.update({
          data: {
            kyc_status: 'SUCCESS',
            jumio_account_id: undefined,
            id_details: [
              {
                id_issuing_country: 'CHL',
                id_subtype: undefined,
                id_type: undefined
              }
            ],
            failure_message: undefined,
            public_address: undefined,
            age: NaN
                 ~~~
          },
          where: {
            id: 7
          }
        })

Argument age: Got invalid value NaN on prisma.updateOneRedemption. Provided Float, expected Int or NullableIntFieldUpdateOperationsInput or Null.
```
Guessing it is interpreting undefined as `NaN` since the parameter is a number and Postgres supports NaN for some numeric fields

This occurs when I had the code as:
```
  async update(
    redemption: Redemption,
    data: {
      kycStatus?: KycStatus;
      jumioAccountId?: string;
      idDetails?: IdDetails[];
      failureMessage?: string;
      publicAddress?: string;
      age?: number;
    },
    prisma?: BasePrismaClient,
  ): Promise<Redemption> {
    const client = prisma ?? this.prisma;

    return client.redemption.update({
      data: {
        kyc_status: data.kycStatus,
        jumio_account_id: data.jumioAccountId,
        id_details: instanceToPlain(data.idDetails),
        failure_message: data.failureMessage,
        public_address: data.publicAddress,
        age: data.age,
      },
      where: {
        id: redemption.id,
      },
    });
  }
```
FWIW prisma handles undefined properly for string fields 🤷